composeApp/src/commonMain/kotlin/com/swmansion/routinetracker/screen/CreateRoutineScreen.kt

```diff
@@ -39,6 +46,18 @@ fun CreateRoutineScreen(
     val uiState by viewModel.uiState.collectAsState()
     val timePickerState = rememberAdaptiveTimePickerState()
 
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    LaunchedEffect(navBackStackEntry) {
+        val handle = navBackStackEntry?.savedStateHandle ?: return@LaunchedEffect
+        val name: String? = handle.get("task_name")
+        if (name != null) {
+            val duration: Int? = handle.get("task_duration")
+            viewModel.addTask(name, duration)
+            handle.remove<String>("task_name")
+            handle.remove<Int>("task_duration")
+        }
+    }
```

Comment:
How about making the two screens share a viewmodel? See [here](https://stackoverflow.com/a/69643625) for an example. This should simplify the data flow: instead of passing things around via `SavedStateHandle`, we’ll be able to make `CreateTaskScreen` call `CreateRoutineViewModel.addTask` directly before popping the back stack.